dev/AppNotifications/NotificationProgressData.cpp

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+
+#include "AppNotificationProgressData.h"
+#include "NotificationProgressData.h"
+#include "AppNotificationUtility.h"
+
+namespace winrt
+{
+    using namespace winrt::Microsoft::Windows::AppNotifications;
+}
+
+namespace ToastABI
+{
+    using namespace ::ABI::Microsoft::Internal::ToastNotifications;
+}
+
+NotificationProgressData::NotificationProgressData(winrt::AppNotificationProgressData const& progressData)
+    : m_progressData(progressData.SequenceNumber())
+{
+    m_progressData.Status(progressData.Status());
+    m_progressData.Title(progressData.Title());
+    m_progressData.Value(progressData.Value());
+    m_progressData.ValueStringOverride(progressData.ValueStringOverride());
+}
+
+STDMETHODIMP NotificationProgressData::get_SequenceNumber(_Out_ unsigned int* value) noexcept
+{
+    auto lock{ m_lock.lock_shared() };
+
+    *value = m_progressData.SequenceNumber();
+
+    return S_OK;
+}
+
+STDMETHODIMP NotificationProgressData::get_Title(_Out_ HSTRING* value) noexcept try
+{
+    auto lock{ m_lock.lock_shared() };
+
+    *value = Microsoft::Windows::AppNotifications::Helpers::safe_make_unique_hstring(m_progressData.Title().c_str()).release();
+
+    return S_OK;
+}
+CATCH_RETURN()
+
+STDMETHODIMP NotificationProgressData::get_Value(_Out_ double* value) noexcept
+{
+    auto lock{ m_lock.lock_shared() };
+
+    *value = m_progressData.Value();
+
+    return S_OK;
+}
+
+STDMETHODIMP NotificationProgressData::get_ValueStringOverride(_Out_ HSTRING* value) noexcept try
+{
+    auto lock{ m_lock.lock_shared() };
+
+    *value = Microsoft::Windows::AppNotifications::Helpers::safe_make_unique_hstring(m_progressData.ValueStringOverride().c_str()).release();
+
+    return S_OK;
+}
+CATCH_RETURN()
+
+STDMETHODIMP NotificationProgressData::get_Status(_Out_ HSTRING* value) noexcept try
+{
+    auto lock{ m_lock.lock_shared() };
+
+    *value = Microsoft::Windows::AppNotifications::Helpers::safe_make_unique_hstring(m_progressData.Status().c_str()).release();
```

Comment:
Since there is nothing fundamentally wrong with the code (no bug or security risk, etc.), should we take things like these as bugs to be addressed after the merge? My concern is that moving this to a new file, means that we'll have to add at least 1 additional file (if it involves only modifying a .h) to this already very large PR.

Reply:
Looks to me that we can address this as a bug fix and get it done after this PR.

Reply:
New issue created to track the work: https://github.com/microsoft/WindowsAppSDK/issues/2207

Reply:
Resolving since we have a bug to fix it after code complete.